### ISSUE TITLE
CVE-2020-14001 Kramdown gem version

### ISSUE THREAD

**Issue by calexander3** [elastic/logstash#12199]
It looks like the version of Kramdown that you are using has a security issue filed. Could you please update the version?

https://security-tracker.debian.org/tracker/CVE-2020-14001
>The kramdown gem before 2.3.0 for Ruby processes the template option inside Kramdown documents by default, which allows unintended read access (such as template="/etc/passwd") or unintended embedded Ruby code execution (such as a string that begins with template="string://<%= `). NOTE: kramdown is used in Jekyll, GitLab Pages, GitHub Pages, and Thredded Forum.

**Comment by calexander3** [comment]
> Note: `kramdown` is a transitive _development_ dependency by way of `logstash-devutils`, and is not included in distributed packages.

Does that mean that even though it is in the `Gemfile.lock` of the installed it is not actually present? I downloaded Logstash from here: https://artifacts.elastic.co/downloads/logstash/logstash-7.9.0.zip.

**Comment by yaauie** [comment]
The only reference to kramdown that is included in the distributed artifact is an adapter within `tilt` (not kramdown itself):

~~~
╭─{ yaauie@limbo:~/src/elastic/logstash (✘ master) }
╰─● unzip -l ~/Downloads/logstash-7.9.0.zip | grep kramdown                              
      514  08-12-2020 00:03   logstash-7.9.0/vendor/bundle/jruby/2.5.0/gems/tilt-2.0.10/lib/tilt/kramdown.rb
[success]
~~~

**Comment by rbrayner** [comment]
My pipeline has also failed when using trivy to inspect the following Logstash image: docker.elastic.co/logstash/logstash-oss .

![image](https://user-images.githubusercontent.com/11684945/93118023-03f2ea00-f696-11ea-92c5-f890570110cc.png)

**Comment by brunocascio** [comment]
My pipeline fails because now is CRITICAL
![image](https://user-images.githubusercontent.com/2397875/99097862-f59e4d00-25b6-11eb-84e8-8a5624358056.png)

**Comment by Joengenduvel** [comment]
Same issue here.
Looks like it needs to be fixed in tilt first, as @yaauie mentioned in https://github.com/elastic/logstash/issues/12199#issuecomment-686916763

I created on issue on the tilt repository as well: https://github.com/rtomayko/tilt/issues/358

**Comment by flauflauf** [comment]
@Joengenduvel The kramdown dependency does not come from tilt, but rather from logstash-devutils. I created an [issue](https://github.com/elastic/logstash-devutils/issues/92) there.

I am still not sure whether this means that it is an issue for logstash as well or not. Is logstash-devutils only used within logstash's build, or is it present in the final image as well?

**Comment by flauflauf** [comment]
Looking inside the logstash image, logstash-devutils does not seem to be among the installed gems:

```
[root@f4a80e189b4a logstash]# gem list

*** LOCAL GEMS ***

bigdecimal (1.2.0)
io-console (0.4.2)
json (1.7.7)
psych (2.0.0)
rdoc (4.0.0)
[root@f4a80e189b4a logstash]# ls /usr/share/gems/gems/
bigdecimal-1.2.0  io-console-0.4.2  json-1.7.7  psych-2.0.0  rdoc-4.0.0
```

Does that mean logstash is unaffected by the CVE?

**Comment by yaauie** [comment]
> Does that mean logstash is unaffected by the CVE?

Yes. As we have previously indicated, kramdown is _not_ included in released Logstash packages and therefore releases of Logstash are _not_ affected by the CVE. It shows up in the `Gemfile.lock` because it is a development dependency.

> Note: `kramdown` is a transitive _development_ dependency by way of `logstash-devutils`, and is not included in distributed packages.

**Comment by dominik-ba** [comment]
since this still comes up in security scans I wonder why you don't update the dev dependencies, too?
